app/grandchallenge/profiles/models.py

```diff
@@ -146,20 +166,58 @@ def get_unsubscribe_link(self, *, subscription_type):
             else:
                 raise ValueError("User has opted out of newsletter emails")
         elif subscription_type == EmailSubscriptionTypes.NOTIFICATION:
-            if self.receive_notification_emails:
+            if (
+                self.notification_email_choice
+                == NotificationEmailOptions.DISABLED
+            ):
+                raise ValueError("User has opted out of notification emails")
+            else:
                 return reverse(
                     "notification-unsubscribe",
                     kwargs={"token": self.unsubscribe_token},
                 )
-            else:
-                raise ValueError("User has opted out of notification emails")
         elif subscription_type == EmailSubscriptionTypes.SYSTEM:
             return None
         else:
             return NotImplementedError(
                 f"Unknown subscription type: {subscription_type}"
             )
 
+    def dispatch_unread_notifications_email(
+        self, *, site, unread_notification_count=None
+    ):
+        self.notification_email_last_sent_at = now()
+        self.save(update_fields=["notification_email_last_sent_at"])
+
+        send_unread_notifications_email(
```

Comment:
`send_unread_notifications_email` doesn't seem to be used anywhere else and in this case should not be used directly. Maybe get rid of the `send_unread_notifications_email` function and put everything it's doing in this new method? Same with `send_new_unread_direct_messages_email`. That way we do not need to make sure that developers only call the dispatch methods.